docs/paper/dev/event-api/custom-events.md

```diff
@@ -0,0 +1,122 @@
+---
+slug: /dev/event-api/custom-events
+---
+
+# Custom Events
+
+Creating custom events is a great way to add functionality to your plugin. 
+This will allow for other people to listen for your custom events and add functionality to your plugin.
+
+## Creating a custom event
+
+To create a custom event, you need to create a class that extends `Event`. Each event has a `HandlerList` that contains all the listeners that are listening for that event.
+
+This list is used to call the listeners when the event is called.
+    
+```java title="PaperIsCoolEvent.java"
+
+public class PaperIsCoolEvent extends Event {
+
+    private static final HandlerList HANDLER_LIST = new HandlerList();
+
+    public static HandlerList getHandlerList() {
+        return HANDLER_LIST;
+    }
+
+    @Override
+    public HandlerList getHandlers() {
+        return HANDLER_LIST;
+    }
```

Comment:
Why is this duplicated? is what someone reading this might ask. I'd add something about that above saying you need both.